### PR TITLE
New 800 logical interconnect

### PR DESCRIPTION
**Pull request by jyothis-gireesan**
### Description
API 800 Support for Interconnects

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11.x + gofmt checks .
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/data_source_logical_interconnect.go [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+			"ethernet_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"category": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created": {
```
Comment (AnushaY1916) [inline oneview/data_source_logical_interconnect.go]: You can remove Created and Modified attributes.

File: oneview/data_source_logical_interconnect.go [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+						},
+						"interconnect_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"mac_refresh_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"modified": {
```
Comment (AnushaY1916) [inline oneview/data_source_logical_interconnect.go]: Same comment as above

File: oneview/data_source_logical_interconnect.go [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+			},
+			"logical_interconnect_group_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
```
Comment (AnushaY1916) [inline oneview/data_source_logical_interconnect.go]: I think name is a required parameter as we are using name as the filter to get the particular resource.

File: oneview/data_source_logical_interconnect.go [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+						},
+					},
+				},
+			},
+			"fusion_domain_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interconnect_map": {
+				Type:     schema.TypeList,
```
Comment (madhav-bharadwaj) [inline oneview/data_source_logical_interconnect.go]: can you check if a typeset needs to be used here? And also same with other places having type as typeList

File: oneview/data_source_logical_interconnect.go [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+	d.Set("eTag", logInt.ETAG)
+	d.Set("fusion_domain_uri", logInt.FusionDomainUri)
+	d.Set("logical_interconnect_group_uri", logInt.LogicalInterconnectGroupUri)
+	d.Set("modified", logInt.Modified)
+	d.Set("name", logInt.Name)
+	d.Set("stacking_health", logInt.StackingHealth)
+	d.Set("state", logInt.State)
+	d.Set("status", logInt.Status)
+	d.Set("type", logInt.Type)
+	d.Set("uri", logInt.URI)
```
Comment (madhav-bharadwaj) [inline oneview/data_source_logical_interconnect.go]: why aren't we setting the other attributes of the resource here?

Comment (jyothis-gireesan) [reply oneview/data_source_logical_interconnect.go]: All attributes are added.

File: oneview/data_source_logical_interconnect.go (hunk 4 above)
Comment (jyothis-gireesan) [reply oneview/data_source_logical_interconnect.go]: MultiLevel schema is not working with TypeSets

**Pull request merged**